std/zip.d

```diff
@@ -291,6 +291,27 @@ final class ZipArchive
     import std.conv : to;
     import std.datetime.systime : DosFileTime;
 
+private:
+    // names are taken directly from the specification
+    // https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT
+    enum centralFileHeaderSignature = [ 0x50, 0x4b, 0x01, 0x02 ];
```

Comment:
This is an array of `int`s:

```d
enum centralFileHeaderSignature = [ 0x50, 0x4b, 0x01, 0x02 ];
pragma(msg, typeof(centralFileHeaderSignature[0])); // int
```

Assigning it to an `ubyte[]` will work, but it would probably be better to fix this anyway.

Reply:
It's a bad idea to use enums for arrays anyway. But as far as I know, the ints would not be a problem, because the compiler notices at the places where this enum is used, that this should be bytes and uses bytes. So there are only ints because you queried them with the pragma. A little bit like the "Uncertainty principle" from physics. Unfortunately I cannot find the quote in the documentation, where I read this.